src/Microsoft.IdentityModel.Xml/Signature.cs

```diff
@@ -91,29 +91,34 @@ public SignedInfo SignedInfo
         /// Verifies the signature over the SignedInfo.
         /// </summary>
         /// <param name="key">the <see cref="SecurityKey"/> to use for cryptographic operations.</param>
+        /// <param name="securityToken">The <see cref="SecurityToken"/> being validated.</param>
+        /// <param name="validationParameters"><see cref="TokenValidationParameters"/> required for validation.</param>
         /// <exception cref="ArgumentNullException"> if <paramref name="key"/> is null.</exception>
         /// <exception cref="XmlValidationException"> if <see cref="SignedInfo"/> null.</exception>
         /// <exception cref="XmlValidationException"> if <see cref="SignedInfo.SignatureMethod"/> is not supported.</exception>
         /// <exception cref="XmlValidationException"> if signature does not validate.</exception>
-        public void Verify(SecurityKey key)
+        public void Verify(SecurityKey key, SecurityToken securityToken, TokenValidationParameters validationParameters)
```

Comment:
Modifying a public method like this would be considered a breaking change. I would add this method in as an overload (same for the other Verify() method as well).

Reply:
Very good point! How should input validation look for the potential null arguments?

Validate algorithm only if the Security token and TVP are present?